unstable/wlr-output-management-unstable-v1.xml

```diff
@@ -0,0 +1,469 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<protocol name="wlr_output_management_unstable_v1">
+  <copyright>
+    Copyright © 2019 Purism SPC
+
+    Permission to use, copy, modify, distribute, and sell this
+    software and its documentation for any purpose is hereby granted
+    without fee, provided that the above copyright notice appear in
+    all copies and that both that copyright notice and this permission
+    notice appear in supporting documentation, and that the name of
+    the copyright holders not be used in advertising or publicity
+    pertaining to distribution of the software without specific,
+    written prior permission.  The copyright holders make no
+    representations about the suitability of this software for any
+    purpose.  It is provided "as is" without express or implied
+    warranty.
+
+    THE COPYRIGHT HOLDERS DISCLAIM ALL WARRANTIES WITH REGARD TO THIS
+    SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND
+    FITNESS, IN NO EVENT SHALL THE COPYRIGHT HOLDERS BE LIABLE FOR ANY
+    SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+    AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+    ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
+    THIS SOFTWARE.
+  </copyright>
+
+  <description summary="protocol to configure output devices">
+    This protocol exposes interfaces to obtain and modify output device
+    configuration.
+
+    Warning! The protocol described in this file is experimental and
+    backward incompatible changes may be made. Backward compatible changes
+    may be added together with the corresponding interface version bump.
+    Backward incompatible changes are done by bumping the version number in
+    the protocol and interface names and resetting the interface version.
+    Once the protocol is to be declared stable, the 'z' prefix and the
+    version number in the protocol and interface names are removed and the
+    interface version number is reset.
+  </description>
+
+  <interface name="zwlr_output_manager_v1" version="1">
+    <description summary="output device configuration manager">
+      This interface is a manager that allows reading and writing the current
+      output device configuration.
+
+      Output devices that display pixels (e.g. a physical monitor or a virtual
+      output in a window) are represented as heads. Heads cannot be created nor
+      destroyed, but they can be enabled or disabled and their properties can be
+      changed. Each head may have one or more available modes.
+
+      Heads are advertised when the output manager is bound, and whenever they
+      appear.
```

Comment:
What about disappearing?

```suggestion
      appear or disappear.
```

Reply:
I wanted to mean "new head objects are created and sent" by "heads are advertised". Also heads cannot be removed when the output manager is bound. Maybe we could add "Heads are removed when they disappear"?

Reply:
Does that mean that a head is only sent when it changes, or every time any head changes? I'm not sure what "disappear" would mean in the former case.

Reply:
Okay, so let's go through all possible situations.

### A head "appears"

A new output is plugged in:

```
zwlr_output_manager_v1.head(new id @42)
zwlr_output_head_v1@42.name("VGA-1")
zwlr_output_head_v1@42.description("A new output i just plugged in")
zwlr_output_head_v1@42.enabled(1)
…
zwlr_output_manager_v1.done(64)
```

### A head changes

For instance a head gets disabled. In this case no head "appears" or "disappears". Thus no head object is created or destroyed.

```
zwlr_output_head_v1@42.enabled(0)
zwlr_output_manager_v1.done(65)
```

### A head "disappears"

The output is unplugged.

```
zwlr_output_head_v1@42.finished()
zwlr_output_manager_v1.done(66)
```

### Or any combination of the above

When multiple changes happen at the same time, they are all grouped before a single `done` event. For instance undocking a laptop will make many outputs go away at the same time.

Reply:
Thanks, this makes sense, and indeed you put notes in head::finished as well as head::enabled. I'm going to be fine with this if there's a note that mentions that not all state needs to be resent at every change.